locale/en/docs/guides/security/index.md

```diff
@@ -0,0 +1,440 @@
+statement: _“Node.js trusts the code being run”_.
+
+**Mitigations**
+
+The `--frozen-intrinsics` flag enables experimental[¹][experimental-features]
+frozen intrinsics, which means all the built-in JavaScript objects and functions
+are recursively frozen.
+Therefore, the following snippet **will not** override the default behavior of
+`Array.prototype.push`
+
+```js
+// eslint-disable-next-line no-extend-native
+Array.prototype.push = function (item) {
+  // overriding the global [].push
+};
+
+// Uncaught:
+// TypeError <Object <Object <[Object: null prototype] {}>>>:
+// Cannot assign to read only property 'push' of object ''
+```
+
+However, it’s important to mention you can still define new globals and replace
+existing globals using `globalThis`
+```console
+> globalThis.foo = 3; foo; // you can still define new globals
+3
+> globalThis.Array = 4; Array; // However, you can also replace existing globals
+4
+```
+
+Therefore, `Object.freeze(globalThis)` can be used to guarantee no globals will
+be replaced.
+
+### Prototype Pollution Attacks (CWE-1321)
+
+Prototype pollution refers to the possibility to modify or inject properties
+into Javascript language items by abusing the usage of _\_proto\__,
+_constructor_, and _prototype_.
+
+Example:
+
+```js
+const a = {"a": 1, "b": 2};
+const data = JSON.parse('{"__proto__": { "polluted": true}}');
+
+const c = Object.assign({}, a, data);
+console.log(c.polluted); // true
+```
+
+This is considered a potential vulnerability inherited from the JavaScript
+language.
+
+**Examples**:
+
+* [CVE-2022-21824][] (Node.js)
+* [CVE-2018-3721][] (3rd Party library: Lodash)
+
+**Mitigations**
+
+* Avoid [insecure recursive merges][], see [CVE-2018-16487][].
+* Implement JSON Schema validations for external/untrusted requests.
+* Create Objects without prototype by using `Object.create(null)`.
+* Freezing the prototype: `Object.freeze(MyObject.prototype)`.
+* Disable the `Object.prototype.__proto__` property using `--disable-proto` flag.
+
+### Uncontrolled Search Path Element (CWE-427)
+
+Node.js loads modules following the [Module Resolution Algorithm][].
+Therefore, it assumes the directory in which a module is requested
+(require) is trusted.
+
+By that, it means the following application behavior is expected.
+Assuming the following directory structure:
+
+* _app/_
+  * _server.js_
+  * _auth.js_
+  * _auth_
+
+If server.js uses `require('./auth')` it will follow the module resolution
+algorithm and load _auth_ instead of _auth.js_.
+
+**Mitigations**
+
+Using the experimental[¹][experimental-features]
+[policy mechanism with integrity checking][] can avoid the above threat.
+For the directory described above, one can use the following `policy.json`
+
+```json
+{
+  "resources": {
+    "./app/auth.js": {
+      "integrity": "sha256-iuGZ6SFVFpMuHUcJciQTIKpIyaQVigMZlvg9Lx66HV8="
+    },
+    "./app/server.js": {
+      "dependencies": {
+        "./auth" : true
```

Comment:
```suggestion
        "./auth" : "./app/auth.js"
```

Idk if we want to keep path search by using `true` here. Right now the policy does a source integrity check but not a location check.